addons/payment_adyen/models/payment_token.py

```diff
@@ -0,0 +1,53 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import _, fields, models
+from odoo.exceptions import ValidationError, UserError
+
+
+class PaymentToken(models.Model):
+
+    _inherit = 'payment.token'
+
+    adyen_shopper_reference = fields.Char(
```

Comment:
Again, I'm not sure about the payment.token model, concerning the security.

Reply:
Handling it too